### PR TITLE
Feature lew 299 add mass generation endpoints in api

### PR DESCRIPTION
**Pull request by filip-prime**
BlueLife API updated - giftcoin links endpoints added. Allow for group/mass generation of gift coin links. All offchain code removed

### CODE REVIEW

File: src/Lykke.blue.Api.AzureRepositories/LykkeSettings/LykkeGlobalSettingsEntity.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.WindowsAzure.Storage.Table;
+namespace Lykke.blue.Api.AzureRepositories.LykkeSettings
+{
+    public class LykkeGlobalSettingsEntity : TableEntity
+    {
+        // ReSharper disable once UnusedMember.Global
+        // used by AutoMapper
+        public bool IsOnMaintenance { get; set; }
```
Comment (NovichikhinAlexey) [inline src/Lykke.blue.Api.AzureRepositories/LykkeSettings/LykkeGlobalSettingsEntity.cs:8]: this settings realy need save to Database? Maybe better save in config from SettingsService?

File: src/Lykke.blue.Api/Controllers/ReferralLinksController.cs [hunk 2]
```diff
@@ -75,20 +91,98 @@ public class ReferralLinksController : BluApiBaseController
...
+        {
+            var serviceRequest = request.ConvertToServiceModel();
+            serviceRequest.RecipientClientId = _requestContext.ClientId;
+
+            var result = await ExecuteRefLinksMethod(p => _referralLinksService.ClaimGiftCoinsWithHttpMessagesAsync(refLinkId, serviceRequest), serviceRequest, request.LogMessage);
             return result;
         }
-        
+
         private async Task<IActionResult> ExecuteRefLinksMethod<TU, T>(Func<TU, Task<HttpOperationResponse<T>>> method, TU request, string logMessage)
```
Comment (NovichikhinAlexey) [inline src/Lykke.blue.Api/Controllers/ReferralLinksController.cs]: really it's bad practic. 
Better in client libraty for referralLinksService make wrapper with good api, Becousr autorest generated not usable interface.
And here ve can use simple and clear logic withpur parce http respince

File: src/Lykke.blue.Api/Controllers/TwitterController.cs [hunk 3]
```diff
@@ -41,11 +38,11 @@ IInspireStreamClient inspireStreamClient
         [ProducesResponseType(typeof(IEnumerable<JObject>), (int)HttpStatusCode.OK)]
         [ProducesResponseType((int)HttpStatusCode.BadRequest)]
         [ProducesResponseType((int)HttpStatusCode.NotFound)]
-        public async Task<IActionResult> GetTweetsJSON([FromBody]TweetsRequestModel model)
+        public async Task<IActionResult> GetTweetsJson([FromBody]TweetsRequestModel model)
         {
-            var result = await _inspireStreamClient.GetAsync(model.CreateReques(model));
+            var result = (await _inspireStreamClient.GetAsync(model.CreateReques(model)))?.ToList();
```
Comment (NovichikhinAlexey) [inline src/Lykke.blue.Api/Controllers/TwitterController.cs]: better use 
var resEnum = await _inspireStreamClient.GetAsync(model.CreateReques(model));
var result = resEnum as TYPE_ELEMENT[] ?? resEnum.ToArray();

File: src/Lykke.blue.Api/Lykke.blue.Api.csproj [hunk 4]
```diff
@@ -37,23 +37,23 @@
...
+    <PackageReference Include="Lykke.Common" Version="3.1.0" />
     <PackageReference Include="Lykke.Common.ApiLibrary" Version="1.0.14" />
-    <PackageReference Include="Lykke.Logs" Version="3.2.2" />
+    <PackageReference Include="Lykke.Logs" Version="3.5.1" />
     <PackageReference Include="Lykke.Service.ClientAccount.Client" Version="1.1.11" />
     <PackageReference Include="Lykke.Service.Pledges.Client" Version="1.0.10-beta" />
     <PackageReference Include="Lykke.Service.Registration" Version="1.0.32-beta9" />
     <PackageReference Include="Lykke.Service.Session" Version="1.0.34-beta67" />
-    <PackageReference Include="Lykke.SettingsReader" Version="1.0.42" />
+    <PackageReference Include="Lykke.SettingsReader" Version="1.1.3" />
```
Comment (NovichikhinAlexey) [inline src/Lykke.blue.Api/Lykke.blue.Api.csproj]: update to 1.1.4 in solution
becouse in 1.1.3 we have critical bug

File: src/Lykke.blue.Api.AzureRepositories/LykkeSettings/LykkeGlobalSettingsEntity.cs (hunk 1 above)
Comment (filip-prime) [reply src/Lykke.blue.Api.AzureRepositories/LykkeSettings/LykkeGlobalSettingsEntity.cs:8]: We don't actually save it in DB, we just read it from DB. IsOnMaintenance flag is a global Lykke settings flag, stored in the Setup table in lkedevmain account. We just read it from there and we don't have control over it. This setting is not meant to be local setting for the Ref Links service. I have just created a repo for reading it.

File: src/Lykke.blue.Api/Controllers/ReferralLinksController.cs (hunk 2 above)
Comment (filip-prime) [reply src/Lykke.blue.Api/Controllers/ReferralLinksController.cs]: Done. API now uses RefLinks service client which has been refactored to return DTOs and ObjectResult and to parse the Http response messages and return accordingly. API is now unaware of the existence of the Autorest client.

File: src/Lykke.blue.Api/Controllers/TwitterController.cs (hunk 3 above)
Comment (filip-prime) [reply src/Lykke.blue.Api/Controllers/TwitterController.cs]: Done.

File: src/Lykke.blue.Api/Lykke.blue.Api.csproj (hunk 4 above)
Comment (filip-prime) [reply src/Lykke.blue.Api/Lykke.blue.Api.csproj]: Done

**Review by NovichikhinAlexey — APPROVED** [review]

**Pull request merged**